### DISCUSSION TITLE
Statically exporting a workflow DAG

### DISCUSSION THREAD

**Discussion by ieguinoa** [nextflow-io/nextflow#2451] [Ideas]
Hi all,

I would like to ask about the possibility of statically parsing the workflow details from a workflow definition in order to obtain something similar to the DAG but without the need to actually run any step and (possibly) without the need to define the inputs. 
Since some of the process definitions can be dependent on inputs then I'm not even sure if this is possible, but my intention is to generate a kind of tree of the tokens (process,channels, etc) of the language from a workflow instance. This data model can be useful to represent the whole workflow, independent of the which path of the tree a specific execution takes. So, my question is mainly, is it possible to do the parsing of the DSL2 language into a kind of syntax tree? Is this done at some point in the code?
Just to get the discussion going, I've briefly tested an attempt to get this information by using the -stub-run parameter, but first adding empty channels to all inputs, but this is definitely a bit of a hacky way and I assume only works for some cases, plus I still dont have an easy way to automatically add empty stub sections to all process definitions in case the workflow does not defines it.

I'm willing to put some work on it, even dive in the code if needed but I'm a bit lost in terms of what is the best approach to get all this information so any input in the correct direction is more than welcome.

Thanks!
Ignacio

**Comment by pditommaso** [comment]
Most of the configuration of NF pipeline comes from the pipeline config file. One simple solution could be relying on the output of the `nextflow config` command, eg. 

```
nextflow config nf-core/rnaseq
```

The drawback of this approach it won't provide the setting defined using the directive in the script files.
